### PR TITLE
[Runtime] Don't cache malloc_default_zone.

### PR DESCRIPTION
**Pull request by mikeash**
Caching the default zone should no longer be necessary.

rdar://102870625

### CODE REVIEW

**Review by Catfish-Man — APPROVED** [review]
Seems fine

**Comment by mikeash** [comment]
@swift-ci please test and merge

**Comment by mikeash** [comment]
@swift-ci please test macos platform

**Comment by yln** [comment]
I like this!

File: stdlib/public/runtime/Heap.cpp [hunk 1]
```diff
@@ -108,13 +97,13 @@ void *swift::swift_slowAllocTyped(size_t size, size_t alignMask,
...
-    void *p;
+    void *p = nullptr;
     // This check also forces "default" alignment to use malloc_memalign().
     if (alignMask <= MALLOC_ALIGN_MASK) {
-      p = malloc_type_zone_malloc(DEFAULT_ZONE(), size, typeId);
+      p = malloc_type_malloc(size, typeId);
     } else {
       size_t alignment = computeAlignment(alignMask);
-      p = malloc_type_zone_memalign(DEFAULT_ZONE(), alignment, size, typeId);
+      malloc_type_memalign(&p, alignment, size, typeId);
```
Comment (yln) [inline stdlib/public/runtime/Heap.cpp:106]: Nit: use `malloc_type_aligned_alloc` to avoid `memalign`'s awkward interface (out parameter).

**Review by yln — APPROVED** [review]

File: stdlib/public/runtime/Heap.cpp (hunk 1 above)
Comment (mikeash) [reply stdlib/public/runtime/Heap.cpp]: And fixed.

**Comment by mikeash** [comment]
@swift-ci please smoke test and merge

**Pull request merged**